File: python/1018.py
class Solution:
    def prefixesDivBy5(self, nums: List[int]) -> List[bool]:
        curr = 0
        n = len(nums)
        result = [False] * n
        for i, num in enumerate(nums):
            curr = (curr * 2 + num) % 5
            if curr == 0:
                result[i] = True
        return result

File: cpp/1018.cpp
class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& nums) {
        int curr = 0;
        int n = nums.size();
        vector<bool> result(n, false);
        for (int i = 0; i < n; i++) {
            curr = (curr * 2 + nums[i]) % 5;
            if (curr == 0) result[i] = true;
        }
        return result;
    }
};